### PR TITLE
Added Open AI Organization ID

### PR DESCRIPTION
**Pull request by marcostalder85**
The API Key not working if you are using a paid account from a organization. So an optional OpenAI-Organization is needed for the cURL header.

### CODE REVIEW

File: Workflow/info.plist [hunk 1]
```diff
@@ -1289,6 +1289,27 @@ Query DALL·E via the `dalle` keyword.
...
+				<string></string>
+				<key>placeholder</key>
+				<string></string>
+				<key>required</key>
+				<false/>
+				<key>trim</key>
+				<true/>
+			</dict>
+			<key>description</key>
+			<string>Get it at https://platform.openai.com/api-keys</string>
```
Comment (vitorgalvao) [inline Workflow/info.plist:1305]: The URL is incorrect, as this is the one for API keys. Also, this option won’t be relevant to most people so it should be further down.

File: Workflow/dalle [hunk 2]
```diff
@@ -127,6 +128,10 @@ function run(argv) {
     })
   ]
 
+  if (organizationId && organizationId.trim() !== "") {
+    taskArguments.push("--header", `OpenAI-Organization: ${organizationId}`);
```
Comment (vitorgalvao) [inline Workflow/dalle:132]: `taskArguments` does not exist.

File: Workflow/dalle [hunk 3]
```diff
@@ -127,6 +128,10 @@ function run(argv) {
     })
   ]
 
+  if (organizationId && organizationId.trim() !== "") {
```
Comment (vitorgalvao) [inline Workflow/dalle:131]: This whole block is inconsistent with the one above, when they’re trying to do the exact same thing.

**Review by vitorgalvao — COMMENTED** [review]
The code has some mistakes and doesn’t follow the functional style, so I’ll prepare a version for testing.

**Comment by vitorgalvao** [comment]
Please try [this version](https://github.com/alfredapp/openai-workflow/files/14670623/ChatGPT.DALL-E.alfredworkflow.zip) and let me know if it works for you. The setting for the organisation ID is at the bottom of the config.

**Comment by marcostalder85** [comment]
It works. Sorry for the fast pulled code.

**Comment by vitorgalvao** [comment]
Thank you for the confirmation. This will be in the next release.

**Pull request closed without merging**